### PR TITLE
Document wait transaction behaviour

### PR DESCRIPTION
**Pull request by mnd999**
Add a note explaining the transaction behaviour of WAIT

### CODE REVIEW

**Review by OliviaYtterbrink — APPROVED** [review]

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/DatabasesTest.scala [hunk 1]
```diff
@@ -228,10 +228,13 @@ class DatabasesTest extends DocumentingTest with QueryStatisticsTestSupport {
...
-          |
-          |The default behaviour is `NOWAIT`, so if no clause is specified the command will return
-          |immediately whilst the action is performed in the background. """.stripMargin)
+          |* `NOWAIT` - Return immediately.""")
+      p(
+        """A command using a `WAIT` clause will automatically commit the current transaction when it executes successfully as the
+          |command needs to run immediately for it to be possible to `WAIT` for it to complete. Any subsequent commands executed will
+          |therefore be performed in a new transaction. This is different to the usual transactional behaviour and for this reason
+          |it is recommended that these commands be run in their own transaction. The default behaviour is `NOWAIT`, so if no clause
+          |is specified the transaction will behave normally and the action is performed in the background post-commit.""".stripMargin)
```
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/DatabasesTest.scala]:
```suggestion
        """A command using a `WAIT` clause will automatically commit the current transaction when it executes successfully, as the
          |command needs to run immediately for it to be possible to `WAIT` for it to complete. Any subsequent commands executed will
          |therefore be performed in a new transaction. This is different to the usual transactional behaviour, and for this reason
          |it is recommended that these commands be run in their own transaction. The default behaviour is `NOWAIT`, so if no clause
          |is specified the transaction will behave normally and the action is performed in the background post-commit.""".stripMargin)
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @mnd999 and @OliviaYtterbrink 

I've pushed a minor edit, and will merge when tests are green 

👍

**Comment by davidoliverSP2** [comment]
@mnd999 It seems this PR might be a bit old and requires a rebase to pick up recent fixes. Would you be able to take a quick look?

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/DatabasesTest.scala (hunk 1 above)
Comment (davidoliverSP2) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/DatabasesTest.scala]:
```suggestion
        """A command using a `WAIT` clause will automatically commit the current transaction when it executes successfully, as the
          |command needs to run immediately for it to be possible to `WAIT` for it to complete. Any subsequent commands executed will
          |therefore be performed in a new transaction. This is different to the usual transactional behavior and for this reason
          |it is recommended that these commands be run in their own transaction. The default behavior is `NOWAIT`, so if no clause
          |is specified, the transaction will behave normally and the action is performed in the background post-commit.""".stripMargin)
```

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @mnd999 

Re-raised my commit as a suggestion....

**Pull request merged**